### ISSUE TITLE
Only top-level schema is strictly validated

### ISSUE THREAD

**Issue by NathanielWroblewski** [deseretbook/classy_hash#9]
I have a very simple schema that follows this basic pattern:

```rb
SCHEMA = {
  collection: [[ { id: Integer } ]]
}.freeze
```

and I notice that even when validating strictly, no errors are raised for objects such as:

```rb
invalid = { collection: [{ id: 1, wutang: true }] }
ClassyHash.validate_strict(invalid, SCHEMA)
# => nil
```

Peeking into the source, I noticed [this](
https://github.com/deseretbook/classy_hash/blob/4f20a9ee8abf9c2991ed7432b126159abef14134/lib/classy_hash.rb#L28), and was wondering if you would consider supporting a deep version of `validate_strict`?

**Comment by mike-bourgeous** [comment]
@NathanielWroblewski What do you think of the API in #15?  It'll be merged for the 0.2.0 release of ClassyHash.

**Comment by NathanielWroblewski** [comment]
This has been resolved, thank you for the help!
